include/RAJA/policy/cuda/MemUtils_CUDA.hpp

```diff
@@ -42,84 +42,66 @@ namespace RAJA
 namespace cuda
 {
 
+namespace detail
+{
 
-//! Allocator for pinned memory for use in basic_mempool
-struct PinnedAllocator {
-
-  // returns a valid pointer on success, nullptr on failure
-  void* malloc(size_t nbytes)
-  {
-    void* ptr;
-    cudaErrchk(cudaHostAlloc(&ptr, nbytes, cudaHostAllocMapped));
-    return ptr;
-  }
-
-  // returns true on success, false on failure
-  bool free(void* ptr)
-  {
-    cudaErrchk(cudaFreeHost(ptr));
-    return true;
-  }
-};
-
-//! Allocator for device memory for use in basic_mempool
-struct DeviceAllocator {
+extern void set_device_allocator(RAJA::Allocator* allocator);
```

Comment:
Backends expose functions to get, set, and reset the allocators. This allows apps to provide their own allocators at startup and remove them when shutting down. Currently this is only exposed for cuda and hip, but a host allocator could be exposed as well.